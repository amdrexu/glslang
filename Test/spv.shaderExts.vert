#version 450 core

#extension GL_AMD_shader_extensions:enable

[[spirv_extinst("AMD_shader_trinary_minmax", 1)]]
float min3(float, float, float);

[[spirv_extinst("AMD_shader_trinary_minmax", 1)]]
vec3 min3(vec3, vec3, vec3);

[[spirv_extinst("AMD_shader_trinary_minmax", 3)]]
int min3(int, int, int);

[[spirv_extinst("AMD_shader_ballot", 4)]]
int mbcntAMD(int);

layout(location = 0) in vec4 fIn;

void main()
{
    float data = min3(fIn.x, fIn.y, fIn.z);
    data += min3(fIn.xyw, vec3(0.0), vec3(2.0)).y;
    data += min3(int(fIn.x), 0, 1);
    data += mbcntAMD(0xFF00FF);

    gl_Position = vec4(data);
}
