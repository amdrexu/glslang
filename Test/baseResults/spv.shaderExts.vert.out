spv.shaderExts.vert
// Module Version 10000
// Generated by (magic number): 80008
// Id's are bound by 60

                              Capability Shader
                              Extension  "SPV_AMD_shader_ballot"
                              Extension  "SPV_AMD_shader_trinary_minmax"
               1:             ExtInstImport  "GLSL.std.450"
              23:             ExtInstImport  "SPV_AMD_shader_trinary_minmax"
              47:             ExtInstImport  "SPV_AMD_shader_ballot"
                              MemoryModel Logical GLSL450
                              EntryPoint Vertex 4  "main" 11 55
                              Source GLSL 450
                              SourceExtension  "GL_AMD_shader_extensions"
                              Name 4  "main"
                              Name 8  "data"
                              Name 11  "fIn"
                              Name 53  "gl_PerVertex"
                              MemberName 53(gl_PerVertex) 0  "gl_Position"
                              MemberName 53(gl_PerVertex) 1  "gl_PointSize"
                              MemberName 53(gl_PerVertex) 2  "gl_ClipDistance"
                              MemberName 53(gl_PerVertex) 3  "gl_CullDistance"
                              Name 55  ""
                              Decorate 11(fIn) Location 0
                              MemberDecorate 53(gl_PerVertex) 0 BuiltIn Position
                              MemberDecorate 53(gl_PerVertex) 1 BuiltIn PointSize
                              MemberDecorate 53(gl_PerVertex) 2 BuiltIn ClipDistance
                              MemberDecorate 53(gl_PerVertex) 3 BuiltIn CullDistance
                              Decorate 53(gl_PerVertex) Block
               2:             TypeVoid
               3:             TypeFunction 2
               6:             TypeFloat 32
               7:             TypePointer Function 6(float)
               9:             TypeVector 6(float) 4
              10:             TypePointer Input 9(fvec4)
         11(fIn):     10(ptr) Variable Input
              12:             TypeInt 32 0
              13:     12(int) Constant 0
              14:             TypePointer Input 6(float)
              17:     12(int) Constant 1
              20:     12(int) Constant 2
              25:             TypeVector 6(float) 3
              28:    6(float) Constant 0
              29:   25(fvec3) ConstantComposite 28 28 28
              30:    6(float) Constant 1073741824
              31:   25(fvec3) ConstantComposite 30 30 30
              38:             TypeInt 32 1
              40:     38(int) Constant 0
              41:     38(int) Constant 1
              46:     38(int) Constant 16711935
              52:             TypeArray 6(float) 17
53(gl_PerVertex):             TypeStruct 9(fvec4) 6(float) 52 52
              54:             TypePointer Output 53(gl_PerVertex)
              55:     54(ptr) Variable Output
              58:             TypePointer Output 9(fvec4)
         4(main):           2 Function None 3
               5:             Label
         8(data):      7(ptr) Variable Function
              15:     14(ptr) AccessChain 11(fIn) 13
              16:    6(float) Load 15
              18:     14(ptr) AccessChain 11(fIn) 17
              19:    6(float) Load 18
              21:     14(ptr) AccessChain 11(fIn) 20
              22:    6(float) Load 21
              24:    6(float) ExtInst 23(SPV_AMD_shader_trinary_minmax) 1(FMin3AMD) 16 19 22
                              Store 8(data) 24
              26:    9(fvec4) Load 11(fIn)
              27:   25(fvec3) VectorShuffle 26 26 0 1 3
              32:   25(fvec3) ExtInst 23(SPV_AMD_shader_trinary_minmax) 1(FMin3AMD) 27 29 31
              33:    6(float) CompositeExtract 32 1
              34:    6(float) Load 8(data)
              35:    6(float) FAdd 34 33
                              Store 8(data) 35
              36:     14(ptr) AccessChain 11(fIn) 13
              37:    6(float) Load 36
              39:     38(int) ConvertFToS 37
              42:     38(int) ExtInst 23(SPV_AMD_shader_trinary_minmax) 3(SMin3AMD) 39 40 41
              43:    6(float) ConvertSToF 42
              44:    6(float) Load 8(data)
              45:    6(float) FAdd 44 43
                              Store 8(data) 45
              48:     38(int) ExtInst 47(SPV_AMD_shader_ballot) 4(MbcntAMD) 46
              49:    6(float) ConvertSToF 48
              50:    6(float) Load 8(data)
              51:    6(float) FAdd 50 49
                              Store 8(data) 51
              56:    6(float) Load 8(data)
              57:    9(fvec4) CompositeConstruct 56 56 56 56
              59:     58(ptr) AccessChain 55 40
                              Store 59 57
                              Return
                              FunctionEnd
